pinot-segment-spi/src/main/java/org/apache/pinot/segment/spi/creator/IndexCreationContext.java

```diff
@@ -463,4 +492,17 @@ public boolean isFst() {
       return _sortedUniqueElementsArray;
     }
   }
+
+  class Timestamp extends Wrapper {
```

Comment:
A timestamp index isn’t really an index, it’s a collection of transformed columns with a range index, so there’s nothing to override, and the implementation is dispersed throughout the codebase anyway so couldn’t be overridden at a single cut point like IndexCreationContexy is designed for, so we don’t need Timestamp here and couldn’t use it.

Reply:
removed